haystack/document_stores/elasticsearch.py

```diff
@@ -2012,27 +2002,59 @@ def _get_vector_similarity_query(self, query_emb: np.ndarray, top_k: int):
 
     def _scale_embedding_score(self, score):
         # adjust approximate knn scores, see https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn
-        if self.embeddings_field_supports_similarity:
-            if self.similarity == "dot_product":
-                if score > 1:
-                    score = score - 1
-                else:
-                    score = -(1 / score - 1)
-            elif self.similarity == "cosine":
-                score = -(1 / score - 2)
-            elif self.similarity == "l2":
-                score = 1 / score - 1
         # adjust exact knn scores, see https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/
```

Comment:
Do you mean to keep both lines of comments?

Reply:
Yes, I'd keep them both as both are relevant here.